spec/lib/sarif/yarn_audit_sarif_spec.rb

```diff
@@ -2,20 +2,34 @@
 require 'json'
 
 describe Sarif::YarnAuditSarif do
+  let(:stub_stdout_failure_2) do
+    File.read('spec/fixtures/yarn_audit/failure-2/stub_stdout.txt')
+  end
+  let(:stub_stderr_failure_2) { "" }
+  let(:stub_status_failure_2) { 26 }
+  let(:stub_stdout_failure_4) do
+    File.read('spec/fixtures/yarn_audit/failure-4/stub_stdout.txt')
+  end
+  let(:stub_stderr_failure_4) { "" }
+  let(:stub_status_failure_4) { 28 }
+
   describe '#parse_issue' do
     let(:scanner) { Salus::Scanners::YarnAudit.new(repository: repo, config: {}) }
     let(:error_id_fail_2) { "1002899" } # was 39 before the great yarn advisory reshuffling of '21
-    before { scanner.run }
 
     context 'scan report with logged vulnerabilites' do
       let(:path) { 'spec/fixtures/yarn_audit/failure-2' }
       let(:repo) { Salus::Repo.new(path) }
       it 'parses information correctly' do
+        status = ProcessStatusDouble.new(stub_status_failure_2)
+        stub_ret = Salus::ShellResult.new(stub_stdout_failure_2, stub_stderr_failure_2, status)
+        allow(scanner).to receive(:run_shell).and_return(stub_ret)
+        scanner.run
         issue = JSON.parse(scanner.report.to_h[:info][:stdout])[0]
         yarn_sarif = Sarif::YarnAuditSarif.new(scanner.report, path)
 
         expect(yarn_sarif.parse_issue(issue)).to include(
-          id: error_id_fail_2,
```

Comment:
can we just update let(:error_id_fail_2) { "1005415" }?  Having the error code abstracted lets you just swap that and reduce the changes needed for

help_url: "https://www.npmjs.com/advisories/#{error_id_fail_2}",
help_url: "https://www.npmjs.com/advisories/1005415",

etc